kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/collections/UniqueTest.kt

```diff
@@ -0,0 +1,61 @@
+package com.sksamuel.kotest.matchers.collections
+
+import io.kotest.assertions.throwables.shouldThrowWithMessage
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.collections.shouldBeUnique
+import io.kotest.matchers.collections.shouldNotBeUnique
+
+class UniqueTest : FunSpec() {
+   init {
+
+      test("List.shouldBeUnique") {
+         shouldThrowWithMessage<AssertionError>("Collection should be unique but contained duplicates of 1") {
+            listOf(1, 1, 2).shouldBeUnique()
+         }
```

Comment:
Should we assert the message as well?

Reply:
We are, the message is the parameter to the assertion.

Reply:
Doh, didn't realize. I'm more used to this pattern :) 

```kotlin
shouldFail {
  listOf(1,1,2).shouldBeUnique()
}.message shouldBe """
  failure message..
"""
```